addons/mail/static/src/core/common/attachment_list.xml

```diff
@@ -12,85 +12,97 @@
+                    t-att-class="{
+                               'g-col-12': env.inChatWindow,
+                               'g-col-3': env.inComposer or attachment.message?.thread?.model === 'discuss.channel' and !env.inChatWindow,
+                               'g-col-4': !env.inComposer and attachment.message?.thread?.model !== 'discuss.channel' and !env.inChatWindow,
+                               'ms-1': isInChatWindowAndIsAlignedRight,
+                               'me-1': !isInChatWindowAndIsAlignedRight,
+                               'o-viewable': attachment.isViewable,
+                               'o-isUploading': attachment.uploading,
+                               }"
+                    t-attf-class="bg-300"
+                    t-att-title="attachment.name ? attachment.name : undefined" role="menu" t-att-aria-label="attachment.name"
+                    t-on-click="() => this.fileViewer.open(attachment, props.attachments)"
+                >
+                    <!-- o_image from mimetype.scss -->
+                    <t t-ref="nonImageMain">
+                        <div class="o-mail-AttachmentCard-image o_image flex-shrink-0 m-1" role="menuitem" aria-label="Preview" t-att-tabindex="-1" t-att-aria-disabled="false" t-att-data-mimetype="attachment.mimetype"/>
+                        <div class="overflow-auto d-flex justify-content-center flex-column px-1">
+                            <div t-if="attachment.name" class="text-truncate" t-out="props.messageSearch?.highlight(attachment.name) ?? attachment.name"/>
+                            <small t-if="attachment.extension" class="text-uppercase" t-esc="attachment.extension"/>
+                        </div>
+                    </t>
+                    <div class="o-mail-AttachmentCard-aside position-relative rounded-end overflow-hidden d-flex flex-grow-1" t-att-class="{ 'o-hasMultipleActions d-flex flex-column': showDelete and !env.inComposer }">
+                        <div t-if="attachment.uploading" class="d-flex justify-content-center align-items-center w-100 h-100" title="Uploading">
+                            <i class="fa fa-spin fa-spinner"/>
+                        </div>
+                        <div t-if="!attachment.uploading and env.inComposer" class="d-flex justify-content-end me-2 align-items-center w-100 h-100 text-primary" title="Uploaded">
+                            <i class="fa fa-check"/>
+                        </div>
+                        <button t-if="showDelete"
+                                class="o-mail-AttachmentCard-unlink btn top-0 align-items-center d-flex w-100 h-100 rounded-0 border-0"
+                                t-att-class="{
+                                      'justify-content-end': !env.inComposer,
+                                      'justify-content-center': env.inComposer,
+                                      }"
```

Comment:
For just 2 opposing conditions, prefer ternary over object syntax. This is less verbose.

If we need to add more conditionals in the future, let's turn ternary syntax into object syntax at that moment.
Let's keep code as brief / verbose as we need right now.